src/pykeen/evaluation/classification_evaluator.py

```diff
@@ -27,19 +30,25 @@ class ClassificationMetricResults(MetricResults):
     metrics = CLASSIFICATION_METRICS
 
     @classmethod
-    def from_scores(cls, y_true, y_score):
+    def from_scores(cls, y_true: np.ndarray, y_score: np.ndarray):
         """Return an instance of these metrics from a given set of true and scores."""
         data = dict()
         for key, metric in CLASSIFICATION_METRICS.items():
+            if y_true.size == 0:
+                logger.warning("Empty y_true for %s?!", key)
```

Comment:
can we make this message a bit more actionable

Reply:
Hm, it should be the same reason as https://github.com/pykeen/pykeen/pull/821, though I am not 100% sure. There we decided to use

```python
logger.debug("Empty ranks. This should only happen during size probing.")
```

Reply:
I reverted this change back to raising an error. The underlying issue was fixed by given access to the "clear intermediate result" logic via a new `clear` method, and using this method in favour of `finalize` in batch size probing.